Library/Homebrew/cmd/readall.rb

```diff
@@ -8,6 +8,25 @@
 
 module Homebrew
   def readall
+    if ARGV.delete("--syntax")
+      ruby_files = Queue.new
+      Dir.glob("#{HOMEBREW_LIBRARY}/{Homebrew,Formula}/**/*.rb").each do |rb|
+        ruby_files << rb
+      end
+
+      workers = (0...Hardware::CPU.cores).map do
+        Thread.new do
+          begin
+            while rb = ruby_files.pop(true)
+              nostdout { system "ruby", "-c", rb }
```

Comment:
En, didn't find anything useful, either. It's petty ruby doesn't offer AST API like Python does.